subsys/bluetooth/mesh/settings.c

```diff
@@ -111,10 +111,12 @@ void bt_mesh_settings_store_schedule(enum bt_mesh_settings_flag flag)
 
 	if (atomic_get(pending_flags) & NO_WAIT_PENDING_BITS) {
 		timeout_ms = 0;
+#ifdef CONFIG_BT_MESH_RPL_STORAGE_MODE_SETTINGS
 	} else if (CONFIG_BT_MESH_RPL_STORE_TIMEOUT >= 0 &&
 		   atomic_test_bit(pending_flags, BT_MESH_SETTINGS_RPL_PENDING) &&
 		   !(atomic_get(pending_flags) & GENERIC_PENDING_BITS)) {
 		timeout_ms = CONFIG_BT_MESH_RPL_STORE_TIMEOUT * MSEC_PER_SEC;
+#endif
```

Comment:
To make this code unconditionally compilable, you can do something like this:
```
#ifdef CONFIG_BT_MESH_RPL_STORE_TIMEOUT
#define RPL_STORE_TIMEOUT CONFIG_BT_MESH_RPL_STORE_TIMEOUT
#else
#define RPL_STORE_TIMEOUT (-1)
#endf
```
Then use IS_ENABLED(CONFIG_BT_MESH_RPL_STORE_TIMEOUT) in the if-statement and replace `CONFIG_BT_MESH_RPL_STORE_TIMEOUT` with `RPL_STORE_TIMEOUT` on line 118.

Reply:
https://github.com/Andrewpini/zephyr/blob/b6dfd557fb5794eff757286dfeedc73b17d18508/subsys/bluetooth/mesh/settings.c#L123